src/Elastic.Apm/Model/Span.cs

```diff
@@ -406,5 +412,23 @@ private Destination DeduceHttpDestination()
 				return null;
 			}
 		}
+
+		public void SetLabel(string key, string value)
```

Comment:
I wonder if we should have only a single `SetLabel(string key, Label value)` method, and delegate the "overloads" to the supported implicit conversions of `Label`. It would alleviate the need to maintain 6 overloads in all places that support setting labels. The negative is the somewhat opaqueness of implicit conversions, with the overloads acting to promote the implicit conversions supported. 

Both the following would work with a single `SetLabel(string key, Label value)` method

```c#
SetLabel("foo", "bar");
```

and

```c#
SetLabel("foo", new Label("bar"));
```

Reply:
Hmm... I personally prefer having baked in types on the API surface. If people see `SetLabel(string, bool)` and the other overloads they'll immediately know what types they can pass in. I'd keep it for now.

Reply:
ok. Definitely more discoverable with overloads. I've maybe been thinking too much in Rust lately!